### DISCUSSION TITLE
Boundary Conditions

### DISCUSSION THREAD

**Discussion by Jwpr-dpr** [firedrakeproject/firedrake#3594] [Firedrake support] [answered]
I'm solving a basic electric potential problem using poisson's equation. In this case, i need to use dirichlet boundary conditions in a square domain. However, I need to set different conditions depending on the zone of the quadrilateral. How can i set different boundary conditions in different zones om the boundary?
My goal is to reproduce the problem worked by Polycarpou in his book "Introduction to finite elements in electromagenitism"

**Comment by colinjcotter** [comment, marked as answer]
See the section of the manual on subdomain IDs.
<https://www.firedrakeproject.org/variational-problems.html#incorporating-boundary-conditions>
Defining variational problems — Firedrake 0+unknown documentation<https://www.firedrakeproject.org/variational-problems.html#incorporating-boundary-conditions>
firedrakeproject.org<https://www.firedrakeproject.org/variational-problems.html#incorporating-boundary-conditions>
[icon.ico]<https://www.firedrakeproject.org/variational-problems.html#incorporating-boundary-conditions>

all the best
cjc

On 27 May 2024, at 21:14, Jhon Wilmer Pino Román ***@***.***> wrote:

﻿

I'm solving a basic electric potential problem using poisson's equation. In this case, i need to use dirichlet boundary conditions in a square domain. However, I need to set different conditions depending on the zone of the quadrilateral. How can i set different boundary conditions in different zones om the boundary?
My goal is to reproduce the problem worked by Polycarpou in his book "Introduction to finite elements in electromagenitism"

**Reply by Jwpr-dpr** [reply]
Thank you so much. Problem solved!!!!
